Update directives-reference.mdx with optional root margin (#6038)
withastro/docs: src/content/docs/en/reference/directives-reference.mdx

<HeavyImageCarousel client:visible />
```

> 🆕 **New Feature - Optional rootMargin:** This feature allows for earlier component hydration in scenarios like slow internet connections or when components alter the page layout after loading. By setting a rootMargin, hydration is initiated when this specified margin around the component enters the viewport, rather than the component itself.
> ```astro
> <HeavyImageCarousel client:visible="200px" />
> ```
> Implementing a rootMargin, such as "200px", is beneficial for reducing layout shifts(CLS) and ensuring components are interactively ready sooner, enhancing the stability and responsiveness of the page.
> 

Reviewer: ```suggestion
`client:visible={rootMargin}` loads and hydrates the component JavaScript when a specified margin (in pixels) around the component enters the viewport, rather than the component itself.

```astro
<HeavyImageCarousel client:visible="200px" />
```

Adding an optional `rootMargin` value can reduce layout shifts (CLS),  allow more time for a component to hydrate on slower internet connections, and make components interactive sooner, enhancing the stability and responsiveness of the page.
```

<HeavyImageCarousel client:visible />
```

#### `client:visible={{rootMargin}}`
<p><Since v="4.1.0" /></p>

> Optionally, pass a value to specify options to pass to the underlying `IntersectionObserver`. Currently, the only option supported is `rootMargin`. When `rootMargin` is specified, the component JavaScript will hydrate when a specified margin (in pixels) around the component enters the viewport, rather than the component itself.

Reviewer: ```suggestion
Optionally, a value for `rootMargin` can be passed to the underlying `IntersectionObserver`. When `rootMargin` is specified, the component JavaScript will hydrate when a specified margin (in pixels) around the component enters the viewport, rather than the component itself.
```
